app/Community/Controllers/MessageThreadsController.php

```diff
@@ -0,0 +1,200 @@
+use App\Community\Actions\CreateMessageThreadAction;
+use App\Community\Actions\DeleteMessageThreadAction;
+use App\Community\Actions\ReadMessageThreadAction;
+use App\Community\Models\Message;
+use App\Community\Models\MessageThread;
+use App\Community\Models\MessageThreadParticipant;
+use App\Http\Controller;
+use App\Site\Models\User;
+use Illuminate\Contracts\View\View;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Http\Request;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Validator;
+
+class MessageThreadsController extends Controller
+{
+    public function index(Request $request): View
+    {
+        $user = $request->user();
+
+        $messageThreads = MessageThread::join('message_thread_participants', 'message_thread_participants.thread_id', '=', 'message_threads.id')
+            ->where('message_thread_participants.user_id', $user->ID)
+            ->whereNull('message_thread_participants.deleted_at');
+        $totalMessages = $messageThreads->count();
+
+        $pageSize = 20;
+        $currentPage = (int) ($request->input('page.number') ?? 1);
+        if ($currentPage < 1) {
+            $currentPage = 1;
+        }
+        $totalPages = (int) (($totalMessages + 19) / 20);
+
+        $messageThreads = $messageThreads
+            ->join('messages', 'messages.id', '=', 'message_threads.last_message_id')
+            ->orderBy('messages.created_at', 'DESC')
+            ->offset(($currentPage - 1) * $pageSize)
+            ->limit($pageSize)
+            ->select(['message_threads.*',
+                DB::raw('messages.author_id AS last_author_id'),
+                DB::raw('messages.created_at AS last_message_at'),
+            ])
+            ->get();
+
+        foreach ($messageThreads as &$messageThread) {
+            $messageThread['other_participants'] = User::withTrashed()
+                ->join('message_thread_participants', 'message_thread_participants.user_id', '=', 'UserAccounts.ID')
+                ->where('message_thread_participants.thread_id', '=', $messageThread->id)
+                ->where('message_thread_participants.user_id', '!=', $user->ID)
+                ->pluck('UserAccounts.User')
+                ->toArray();
+            $messageThread['num_unread'] = MessageThreadParticipant::where('user_id', $user->ID)
+                ->where('thread_id', $messageThread->id)
+                ->value('num_unread');
+        }
+
+        return view('community.components.message.index', [
+            'messages' => $messageThreads,
+            'totalPages' => $totalPages,
+            'currentPage' => $currentPage,
+            'unreadCount' => $request->user()->UnreadMessageCount,
+            'totalMessages' => $totalMessages,
+        ]);
+    }
+
+    public function show(Request $request, int $threadId): View
+    {
+        $thread = MessageThread::firstWhere('id', $threadId);
+        if (!$thread) {
+            abort(404);
+        }
+
```

Comment:
To make use of route model binding (see suggested adjustments to RouteServiceProvider)
`public function show(Request $request, Message $messageThread): View`

This can be removed, route model binding takes care of fetching and 404 failovers:
```php
$thread = MessageThread::firstWhere('id', $threadId);
if (!$thread) {
    abort(404);
}
```

Reply:
That doesn't seem to work. Did I name my resource wrong?
```
Route::resource('message-thread', MessageThreadController::class)->only(['show', 'destroy']);
```

```
public function show(Request $request, MessageThread $thread): View
{
    print_r($thread->toArray());
    exit;
```
prints an empty array.

Reply:
I think it's the parameter name that has to match so the autowiring of route model binding can do its job via reflection. I think it actually looks at parameter names.

Does this work? `MessageThread $messageThread`